addons/web_editor/models/ir_attachment.py

```diff
@@ -60,3 +62,53 @@ def _get_media_info(self):
         """Return a dict with the values that we need on the media dialog."""
         self.ensure_one()
         return self._read_format(['id', 'name', 'description', 'mimetype', 'checksum', 'url', 'type', 'res_id', 'res_model', 'public', 'access_token', 'image_src', 'image_width', 'image_height', 'original_id'])[0]
+
+    def _get_html_fields(self):
+        return [('ir.ui.view', 'arch_db', [('type', '=', 'qweb')])]
+
+    def _is_used_in_html_field(self):
+        """Returns a model where this attachment is used, or a dict with model names or False."""
+        self.ensure_one()
+        # Keep significant part of attachment url
+        url = self.local_url
+        is_default_local_url = url.startswith('/web/image/%s?' % self.id)
+        if is_default_local_url:
+            url = ('/web/image/%s' if self.image_src else '/web/content/%s') % self.id
+
+        # in-document URLs are html-escaped, a straight search will not
+        # find them
+        url = tools.html_escape(url)
+        big_models = []
+        sudo_models = []
+        for model_name, field_name, domain in self._get_html_fields():
+            if '%s,%s' % (model_name, field_name) not in FULL_SCAN_FIELDS:
+                if model_name in big_models:
+                    continue
+                if self.env[model_name].sudo().with_context(active_test=False).search_count([]) > SCAN_MAX_COUNT:
+                    big_models.append(model_name)
+                    continue
+            if is_default_local_url:
+                domain.extend([
+                    "|", "|", "|", "|",
+                    (field_name, 'like', '"%s-' % url),
+                    (field_name, 'like', "'%s-" % url),
+                    (field_name, 'like', '"%s?' % url),
+                    (field_name, 'like', "'%s?" % url),
+                ])
+            domain.extend([
+                "|",
+                (field_name, 'like', '"%s"' % url),
+                (field_name, 'like', "'%s'" % url),
+            ])
+            matches = self.env[model_name].with_context(active_test=False).search(domain)
+            if matches:
+                return matches
```

Comment:
I am not sure about this, it means that if we have a match on a modal, we are going to shortcircuit and show the matches on that model only, ignoring all other models.
So people will have to handle the matches (adapting the url), then open again to see if any other match, then again, again, until no matches left.

Since we already take care of the perf through the max search limit, this doesn't seem right.

If my attachment is used in a product desc, 2 blogs, 1 page and 1 SO line description (for which I don't have access), regardless of the order those fields are retrieved, I'd except to see:
- This is used in 1 product desc
- This is used in 2 blogs
- This is used in 1 page
- This is used in 1 SO but you don't have access to this model.

What do you think?

Reply:
I did it initially to reduce the number of queries and I thought doing it like this would make it "stable enough" - but it seems it is not the case anyway given your next comment - so it must be changed.
IMO user-wise any "you do not know what you might break" could replace individual records because not all of them can be fixed anyway. But indeed, on a big DB it would always give that message for any delete - so it's not really on an option I guess.
TL;DR: I agree we should do what you suggest.